actions/submit-signing-request/task.ts

```diff
@@ -231,32 +232,35 @@ export class Task {
             }
         });
 
-        const targetFilePath = path.join(process.env.GITHUB_WORKSPACE as string, this.signedArtifactDestinationPath)
+        const targetDirectory = this.resolveOrCreateDirectory(this.outputArtifactDirectory);
 
-        // make sure that the target directory exists
-        const targetDir = path.dirname(targetFilePath);
-        if (!fs.existsSync(targetDir)) {
-            core.info(`Directory ${targetDir} does not exist, and will be created`);
-            fs.mkdirSync(targetDir, { recursive: true });
-        }
+        core.info(`The signed artifact is being downloaded from SignPath and will be saved to ${targetDirectory}`);
 
-        core.info(`The signed artifact is being downloaded from SignPath and will be saved to ${targetFilePath}`);
-        const writer = fs.createWriteStream(targetFilePath)
+        // save the signed artifact to temp ZIP file
+        const tmpZipFile = path.join(targetDirectory, '__signpath_signed_artifact_tmp.zip');
+        const writer = fs.createWriteStream(tmpZipFile)
         response.data.pipe(writer);
         await new Promise((resolve, reject) => {
-            writer.on('finish', resolve)
-            writer.on('error', reject)
+             writer.on('finish', resolve)
+             writer.on('error', reject)
         });
 
-        core.info("The signed artifact has been successfully downloaded from SignPath");
-        return targetFilePath;
-    }
+        // unzip temp ZIP file to the targetDirectory
+        const zip = new nodeStreamZip.async({ file: tmpZipFile });
+        await zip.extract(null, targetDirectory);
 
-    async logArtifactFileStat(artifactPath: string) {
-        await fs.stat(artifactPath, (err, stats) => {
-            const size = fileSize.partial({base: 2, standard: "jedec"});
-            core.info("File size: " + size(stats.size));
-        });
+        // delete temp ZIP file
+        fs.unlinkSync(tmpZipFile);
+
+        core.info(`The signed artifact has been successfully downloaded from SignPath and extracted to ${targetDirectory}`);
     }
 
+    resolveOrCreateDirectory(relativePath:string): string {
+        const absolutePath = path.join(process.env.GITHUB_WORKSPACE as string, relativePath)
+        if (!fs.existsSync(absolutePath)) {
+            core.info(`Directory ${absolutePath} does not exist, and will be created`);
```

Comment:
Without the comma, so it should be "... does not exist and will be created"

Reply:
Also, should we put values in quotes? like Directory `"${absolutePath}"` - same for the other logs?

Reply:
fixed